### PR TITLE
openbangla-keyboard: init at 2.0.0

### PR DESCRIPTION
**Pull request by hqurve**
###### Description of changes

I've temporarily added a Cargo.lock as to not interfere with the build process too much

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
@ofborg build openbangla-keyboard

**Comment by winterqt** [comment]
@ofborg build libsForQt5.openbangla-keyboard

**Review by winterqt — CHANGES_REQUESTED** [review]
Please follow the guidelines outlined in the [Qt section of the manual](https://nixos.org/manual/nixpkgs/stable/#sec-language-qt).

Notably:
- Don't use the top-level Qt `mkDerivation`, add `qtbase` and the wrapper hook manually.
- Put this in `all-packages`, not `qt5-packages`.

File: pkgs/applications/misc/openbangla-keyboard/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    hash = "sha256-z63bsrxhUudb8O29genHmcOy4pzCO+KzadUc7Hs0ULY=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openbangla-keyboard/default.nix:31]:
```suggestion
    rustPlatform.cargo
    rustPlatform.rustc
    rustPlatform.cargoSetupHook
```

Comment (hqurve) [reply pkgs/applications/misc/openbangla-keyboard/default.nix:31]: I don't think `rustPlatform` has these two attributes, [`makeRustPlatform`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/rust/make-rust-platform.nix) does not list them and tab complete in the repl doesnt either

```
nix-repl> a = import ./. { } 

nix-repl> a.rustPlatform.
a.rustPlatform.bindgenHook        a.rustPlatform.cargoCheckHook     a.rustPlatform.fetchCargoTarball  a.rustPlatform.rust
a.rustPlatform.buildRustPackage   a.rustPlatform.cargoInstallHook   a.rustPlatform.importCargoLock    a.rustPlatform.rustLibSrc
a.rustPlatform.cargoBuildHook     a.rustPlatform.cargoSetupHook     a.rustPlatform.maturinBuildHook   a.rustPlatform.rustcSrc

```

**Comment by hqurve** [comment]
Changing to init at 2.0.0 since the only differences to master is packaging and readmes. https://github.com/OpenBangla/OpenBangla-Keyboard/compare/2.0.0...master

File: pkgs/applications/misc/openbangla-keyboard/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/openbangla-keyboard/default.nix:32]: They are under rustPlatform.rust.

Comment (hqurve) [reply pkgs/applications/misc/openbangla-keyboard/default.nix:32]: thanks

**Pull request merged**